### PR TITLE
Shipping Settings: Remove zone method defaults

### LINKED ISSUE #40738: Shipping Settings: Remove defaults when configuring a new method
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**
When configuring a new Shipping Zone Method, remove all the defaults and disable the "Create" button

image

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fixes https://github.com/woocommerce/woocommerce/issues/40738

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a new Shipping Zone `/wp-admin/admin.php?page=wc-settings&tab=shipping`
2. Add new Methods Free Shipping, Local Pickup, and Flate Rate
3. See defaults match TgDI6pJuBmnPyZ810B7uzX-fi
**Note - ** Selects don't have a placeholder in the [WooCommerce Settings API](https://github.com/woocommerce/woocommerce/blob/trunk/docs/extension-development/settings-api.md), so those will need to be the default selection instead.
4. Be sure to play around with the validation that enables/disables the create/save button

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

Fix already implemented behaviour

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-zone-methods.php [hunk 1]
```diff
@@ -148,7 +148,7 @@
 				</article>
 				<footer>
 					<div class="inner">
-						<button id="btn-ok" data-status='{{ data.status }}' class="button button-primary button-large">
+						<button id="btn-ok" data-status='{{ data.status }}' disable='{{ data.status === "new" ? "disabled" : "" }}' class="button button-primary button-large {{ data.status === 'new' ? 'disabled' : '' }}">
```
Comment (ilyasfoo) [inline plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-zone-methods.php]: Typo perhaps, it seems like you can still click on `Create` despite it looks like disabled.

```suggestion
						<button id="btn-ok" data-status='{{ data.status }}' disabled='{{ data.status === "new" ? "disabled" : "" }}' class="button button-primary button-large {{ data.status === 'new' ? 'disabled' : '' }}">
```

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
It seems like Flat rate default still has `0` as the cost:

image

Everything else tested well

**Comment by psealock** [comment]
Thanks for the review @ilyasfoo 

Indeed, the default is `0` - Here is the design for reference. I'm handling currency symbols and formatting separately in https://github.com/woocommerce/woocommerce/pull/40816

image

File: plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-zone-methods.php (hunk 1 above)
Comment (psealock) [reply plugins/woocommerce/includes/admin/settings/views/html-admin-page-shipping-zone-methods.php]: Whoops, nice catch

**Review by ilyasfoo — APPROVED** [review]
> Indeed, the default is 0 - Here is the design for reference. I'm handling currency symbols and formatting separately in https://github.com/woocommerce/woocommerce/pull/40816

Ah, gotcha. The color in design looked a bit grey-ish, I thought it was meant as a placeholder instead of a default value.

In that case, LGTM! 🚢

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀

**Pull request merged**